app/questionnaire/value_source_resolver.py

```diff
@@ -88,25 +88,29 @@ def _get_answer_value(
     def _resolve_list_item_id_for_value_source(
         self, value_source: Mapping
     ) -> str | None:
-        list_item_id: str | None = None
-
         if list_item_selector := value_source.get("list_item_selector"):
             if list_item_selector["source"] == "location":
                 if not self.location:
                     raise InvalidLocationException(
                         "list_item_selector source location used without location"
                     )
+                # Type ignore: the identifier is a string, same below
+                return getattr(self.location, list_item_selector["identifier"])  # type: ignore
 
-                list_item_id = getattr(self.location, list_item_selector["identifier"])
-
-            elif list_item_selector["source"] == "list":
-                list_item_id = getattr(
+            if list_item_selector["source"] == "list":
+                return getattr(  # type: ignore
                     self.list_store[list_item_selector["identifier"]],
                     list_item_selector["selector"],
                 )
 
-        if list_item_id:
-            return list_item_id
+        if value_source["source"] == "supplementary_data":
+            return (
+                self.list_item_id
+                if self.supplementary_data_store.is_data_repeating(
+                    value_source["identifier"]
+                )
+                else None
+            )
```

Comment:
Any reason the behaviour was changed?. Previously it would get to the final `if` even when one of these resolved to None, whereas, with this change it will return None and not get to the final check.

Reply:
I changed it as I don't think they can resolve to None, a `list_item_selector` must have an identifier that is a non-empty string, so I made it consistent with the pattern I added for `supplementary_data` which does need to return None if its None, thinking this would help readability too.  Can adjust if I've missed a case though

Reply:
At the time I was thinking that it was not possible for that path to do anything different... im questioning it now..
If value source was location, then identifier has to be `to_list_item_id` or `list_item_id` and if the getattr returns None, and it then goes into the final check, it will fail `is_repeating_answer` so return None anyway, and I think similarly for the list case, so functionally I believe the behaviour is consistent

Reply:
Yes, I think it would be the same. 👍